lib/modules/manager/npm/update/dependency/index.ts

```diff
@@ -76,20 +90,21 @@ function replaceAsString(
     parsedContents[depType]![depName] = newValue;
   }
   // Look for the old version number
-  const searchString = `"${oldValue}"`;
-  let newString = `"${newValue}"`;
+  const searchString = getStrValue(isYaml, oldValue);
+  let newString = getStrValue(isYaml, newValue);
 
   const escapedDepName = escapeRegExp(depName);
   const patchRe = regEx(`^(patch:${escapedDepName}@(npm:)?).*#`);
   const match = patchRe.exec(oldValue);
   if (match && depType === 'resolutions') {
     const patch = oldValue.replace(match[0], `${match[1]}${newValue}#`);
     parsedContents[depType]![depName] = patch;
-    newString = `"${patch}"`;
+    newString = getStrValue(isYaml, patch);
```

Comment:
In theory, this could result in us going from quoted to non-quoted or vice versa? I wonder if this would break things if you go from a quoted simple string to a non-simple string. I think you'd get doubled quotes.

Reply:
theorically this is only for yaml files, And since any yaml generator usually tends to remove quotes when are not needed, keeping the quotes always will generate constant git changes any time that you run pnpm install or similar stuff, as the new one will drop the quotes.

Reply:
But it's entirely possible that someone will have a "simple" value in their yaml wrapped in quotes (you cannot assume that the quotes will be removed/non-present). And then update to a non-simple value. I think that from `1` to `1.2.3` is an example (e.g. dependency pinning). Please add a test case for that to check

Reply:
I realise while fixing the other comments, that in fact, for versioning the only part which are necessary is for the dependency name, I changed everything in the case of the yaml to search without the quotes, and replace without the quotes, that should keep it as the same than before. (no add or remove quotes)

For your example 1 and 1.2.3 both cases in my old code was without quotes.